CefSharp.BrowserSubprocess.Core/ConcurrentObjectRegistry.cpp

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright © 2010-2015 The CefSharp Project. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#include "stdafx.h"
+#include "ConcurrentObjectRegistry.h"
+
+using namespace System::Threading;
+
+namespace CefSharp
+{
+    namespace Internals
+    {
+        generic <typename T>
+        int64 ConcurrentObjectRegistry<T>::RegisterObject(T entry)
+        {
+            int64 objectId = Interlocked::Increment(_lastId);
+            _entries->TryAdd(objectId, entry);
+            return objectId;
+        }
+        generic <typename T>
+        bool ConcurrentObjectRegistry<T>::TryGetObject(int64 id, T% entry)
+        {
+            return _entries->TryGetValue(id, entry) && entry != nullptr;
```

Comment:
How exactly does `null` get added to a key? Shouldn't a key be removed if the object is `null`?